compatibility_lib/compatibility_lib/package_crawler_static.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ast
+import importlib
+import os
+
+class PackageNotFound(Exception):
+    pass
+
+def get_package_location(pkgname):
+    """returns a dict containing info about modules, classes, functions"""
+    spec = importlib.util.find_spec(pkgname)
+    if spec is None:
+        errmsg = ('Could not find "%s". Please make sure that '
+                  'this package is installed.' % pkgname)
+        raise PackageNotFound(errmsg)
+    locations = [l for l in spec.submodule_search_locations]
+    root_dir = locations[0]
+    return root_dir
+
+def get_package_info(root_dir):
+    info = {}
+    info['modules'] = {}
+    subpackages = []
+    for name in os.listdir(root_dir):
+        if name.startswith('_'):
+            continue
+        elif name.endswith('.py'):
```

Comment:
You might want to skip all the test files, which looks like `test_*.py` or `*_test.py`.

Reply:
done